src/infer_request.cc

```diff
@@ -442,6 +455,16 @@ InferenceRequest::RespondIfError(
           std::move(response), TRITONSERVER_RESPONSE_COMPLETE_FINAL, status),
       (request->LogRequest() + "failed to send error response").c_str());
 
+  request->ReportStatistics(
+      request->model_raw_->MetricReporter().get(),
+      false, // success: Indicates the operation did not succeed
+      0, // compute_start_ns: Start time of the compute operation, 0 because there was an error before computation
+      0, // compute_input_end_ns: End time of input processing, 0 because there was an error before input processing
+      0, // compute_output_start_ns: Start time of output processing, 0 because there was an error before output processing
+      0, // compute_end_ns: End time of the compute operation, 0 because there was an error before computation ended
+      reason // reason: The specific reason for the failure
+  );
```

Comment:
No need for full descriptions here, just mentioning which vars they refer to. ex: https://github.com/triton-inference-server/core/blob/34d2650598ba394c75c368de99d39462d596f79a/src/dynamic_batch_scheduler.cc#L262

```suggestion
  // compute times set to zero because there was an error
  request->ReportStatistics(
      request->model_raw_->MetricReporter().get(),
      false /* success */,
      0 /* compute_start_ns */,
      0 /* compute_input_end_ns */, 
      0 /* compute_output_start_ns */,
      0 /* compute_end_ns */,
      reason
  );
```

Reply:
Also please run the pre-commit hooks so all the style stuff will automatically be linted